quickwit/quickwit-search/src/search_response_rest.rs

```diff
@@ -69,16 +73,13 @@ impl TryFrom<SearchResponse> for SearchResponseRest {
                     })
                 })
                 .transpose()?;
-            Result::<_, SearchError>::Ok((document, snippet_opt))
-        });
-        let (hits, snippets) = itertools::process_results(hits_with_snippet_iter, |iter| {
-            iter.unzip::<_, _, Vec<_>, Vec<_>>()
-        })?;
+            snippet_opts.push(snippet_opt);
+        }
```

Comment:
Let's test this function to avoid regressions in our output format as it happened before.

Reply:
Not sure what you mean but the output with the snippet is tested.

Reply:
This is not a very comprehensive test.